### ISSUE TITLE
False postive for undefined variable

### ISSUE THREAD

**Issue by bgedik** [PyCQA/pylint#2418]
class Parent(object):
        class Child(object):
            NAME = "foo"
        CHILD_LAMBDAS = [lambda: Parent.Child.NAME]

    print Parent.CHILD_LAMBDAS[0]()

### Steps to reproduce
1. Run pylint on the code above

### Current behavior
``[E0602(undefined-variable), Parent.<lambda>] Undefined variable 'Parent'``

### Expected behavior
No errors

### pylint --version output
pylint 1.9.2

**Comment by HoffmannP** [comment]
Same here:

```python
    return lambda y, p: \
        lambda filename, stream: \
            requests.get(
                url_template % (y, p, filename),
                cookies=cookies,
                stream=stream)
```
### Current behavior
`E: 27,32: Undefined variable 'y' (undefined-variable)`
`E: 27,35: Undefined variable 'p' (undefined-variable)`

### pylint --version output
pylint 1.8.3

**Comment by PCManticore** [comment]
Thanks for reporting a bug! I can only reproduce this with 1.9.X, it seems to be working fine in 2.X, so we'll most likely need to backport the fix back to 1.9 branch.

**Comment by PCManticore** [comment]
Closing this as 1.9.X is no longer maintained.
